### PR TITLE
[AKS] Improve messages for upgrade command

### PR DESCRIPTION
**Pull request by lalo**
- Inform users of specific behaviours when they try to upgrade a cluster
to the current version of that cluster.

Skipped the HISTORY.rst since this is only logging, can add if prompted by maintainers (or anyone who disagrees).

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by lalo** [comment]
@mboersma adding you since you've done some work here before, also adding @jluk @yangl900

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py [hunk 1]
```diff
@@ -1732,6 +1732,16 @@ def aks_scale(cmd, client, resource_group_name, name, node_count, nodepool_name=
 
 def aks_upgrade(cmd, client, resource_group_name, name, kubernetes_version, no_wait=False, **kwargs):  # pylint: disable=unused-argument
     instance = client.get(resource_group_name, name)
+
+    if instance.kubernetes_version == kubernetes_version:
+        if instance.provisioning_state == "Succeeded":
+            logger.warning("Your cluster is already on version %s and in a non-failed cluster state. No operations "
```
Comment (mboersma) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py]: "non-failed" sounds a little awkward to me. Maybe reword it like this?

"The cluster is already on version %s and is not in a failed state. No operations "
"will occur when upgrading to the same version if the cluster is not in a failed state."

Comment (jluk) [reply src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py]: Thanks @mboersma agree on change

Comment (lalo) [reply src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py]: done

**Comment by mboersma** [comment]
@tjprescott could this get into the next CLI release?
